pandas/tests/frame/test_query_eval.py

```diff
@@ -1037,63 +1017,29 @@ def test_query_string_scalar_variable(self, parser, engine):
         assert_frame_equal(e, r)
 
 
-class TestDataFrameEvalNumExprPandas(object):
-
-    @classmethod
-    def setup_class(cls):
-        cls.engine = 'numexpr'
-        cls.parser = 'pandas'
-        tm.skip_if_no_ne()
+class TestDataFrameEvalWithFrame(object):
 
     def setup_method(self, method):
         self.frame = DataFrame(randn(10, 3), columns=list('abc'))
 
     def teardown_method(self, method):
         del self.frame
 
-    def test_simple_expr(self):
-        res = self.frame.eval('a + b', engine=self.engine, parser=self.parser)
+    def test_simple_expr(self, parser, engine):
+        res = self.frame.eval('a + b', engine=engine, parser=parser)
         expect = self.frame.a + self.frame.b
         assert_series_equal(res, expect)
 
-    def test_bool_arith_expr(self):
-        res = self.frame.eval('a[a < 1] + b', engine=self.engine,
-                              parser=self.parser)
+    def test_bool_arith_expr(self, parser, engine):
+        res = self.frame.eval('a[a < 1] + b', engine=engine, parser=parser)
         expect = self.frame.a[self.frame.a < 1] + self.frame.b
         assert_series_equal(res, expect)
 
-    def test_invalid_type_for_operator_raises(self):
+    def test_invalid_type_for_operator_raises(self, parser, engine):
         df = DataFrame({'a': [1, 2], 'b': ['c', 'd']})
         ops = '+', '-', '*', '/'
         for op in ops:
             with tm.assert_raises_regex(TypeError,
                                         "unsupported operand type\(s\) "
                                         "for .+: '.+' and '.+'"):
-                df.eval('a {0} b'.format(op), engine=self.engine,
-                        parser=self.parser)
-
-
-class TestDataFrameEvalNumExprPython(TestDataFrameEvalNumExprPandas):
-
-    @classmethod
-    def setup_class(cls):
-        super(TestDataFrameEvalNumExprPython, cls).setup_class()
-        cls.engine = 'numexpr'
-        cls.parser = 'python'
-        tm.skip_if_no_ne(cls.engine)
-
-
-class TestDataFrameEvalPythonPandas(TestDataFrameEvalNumExprPandas):
-
-    @classmethod
-    def setup_class(cls):
-        super(TestDataFrameEvalPythonPandas, cls).setup_class()
```

Comment:
The call to the super ``setup_class`` method inadvertently skipped this class of tests on systems that did not have ``numexpr`` even though that isn't required for these. Fixed with commit